tests/test_x509.py

```diff
@@ -3327,6 +3327,20 @@ def test_init_bad_value(self):
                 b'bytes'
             )
 
+    def test_init_bad_country_code_value(self):
+        with pytest.raises(ValueError):
+            x509.NameAttribute(
+                NameOID.COUNTRY_NAME,
+                u'United States'
+            )
+
+        # unicode string of length 2, but > 2 bytes
+        with pytest.raises(ValueError):
+            x509.NameAttribute(
+                NameOID.COUNTRY_NAME,
+                u'\U0001F37A\U0001F37A'
```

Comment:
I'm losing my mind, how does this not explode on `value.encode("ascii")` that string is distinctly not ascii.